localstack/services/dynamodb/dynamodb_listener.py

```diff
@@ -730,6 +755,17 @@ def dynamodb_extract_keys(item, table_name):
     return result
 
 
+def dynamodb_get_table_stream_specification(table_name):
+    stream_specification = {}
+    ddb_client = aws_stack.connect_to_service('dynamodb')
+    try:
+        stream_specification = ddb_client.describe_table(TableName=table_name)['Table'].get('StreamSpecification')
```

Comment:
nit: Since we're using `stream_specification` only in a single place, we could simplify to:
```
def dynamodb_get_table_stream_specification(table_name):
    ddb_client = aws_stack.connect_to_service('dynamodb')
    try:
        return ddb_client.describe_table(TableName=table_name)['Table'].get('StreamSpecification')
    except Exception as e:
        LOGGER.info('Unable to get stream specification for table "%s": %s %s' % (
            table_name, e, traceback.format_exc()))
        raise e
```

Also, querying the table schema for each `PutItem`/`UpdateItem` operation can have severe performance impacts. Let's better make use of `get_table_schema(..)` here (which internally uses the cached values from `SCHEMA_CACHE`).